org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/environments/ExecutionEnvironmentDescription.java

```diff
@@ -187,18 +184,35 @@ public final class ExecutionEnvironmentDescription {
 	/**
 	 * Execution environment description properties
 	 */
-	private Map<String, String> fProperties = null;
+	private final Map<String, String> fProperties;
 
 	/**
-	 * Creates an execution environment description based on the properties defined in the given
-	 * execution environment description file. The format of the file is defined by
-	 * <code>http://wiki.eclipse.org/Execution_Environment_Descriptions</code>.
+	 * Creates an execution environment description based on the properties defined in the given execution environment description file. The format of
+	 * the file is defined by <code>https://wiki.eclipse.org/Execution_Environment_Descriptions</code>.
 	 *
 	 * @param eeFile execution environment description file
 	 * @throws CoreException if unable to read or parse the file
 	 */
 	public ExecutionEnvironmentDescription(File eeFile) throws CoreException {
-		initProperties(eeFile);
+		this(loadProperties(eeFile), null);
+	}
+
+	/**
+	 * Creates an execution environment description based on the given execution environment description properties. The contained properties are
+	 * defined in <code>https://wiki.eclipse.org/Execution_Environment_Descriptions</code>.
+	 *
+	 * @param eeProperties execution environment description properties
+	 * @throws CoreException if unable to read or parse the file
+	 * @since 3.21
+	 */
+	public ExecutionEnvironmentDescription(Map<String, String> eeProperties) {
+		this(new LinkedHashMap<>(eeProperties), null); // defensive copy
+	}
+
+	private ExecutionEnvironmentDescription(Map<String, String> eeProperties, @SuppressWarnings("unused") Object internal) {
+		fProperties = eeProperties; // has to be set before resolveHome() is called
```

Comment:
better pass eeProperties,.get(EE_HOME) to resolveHome and create a new Map for fProperties. Then the code does not depend on comments

Reply:
Done.